### PR TITLE
 Disable the dimension selection dropdown after restoring a cohort 

### PR DESCRIPTION
**Pull request by ewelinagr**
Fixes [TMT-862](https://jira.thehyve.nl/browse/TMT-862) - when restoring a cohort, the dimension button should be locked.

Additionally, remove unused code in the cohort service.

### CODE REVIEW

File: src/app/services/cohort.service.ts [hunk 1]
```diff
@@ -80,12 +80,6 @@ export class CohortService {
     this.updateCountsWithCurrentCohort();
   }
 
-  clearAll(): Promise<any> {
```
Comment (gijskant) [inline src/app/services/cohort.service.ts]: So, `clearAll` in `src/app/services/mocks/cohort.service.mock.ts` can also be removed?

**Comment by gijskant** [comment]
Only works for the top level now. If I create a query "Select all biomaterials for patients with gender 'Female'", the dimension dropdown is disabled on the inner query. However, when saving and restoring that query, the dropdown is not disabled.
Screenshot example biomaterial query

File: src/app/services/cohort.service.ts (hunk 1 above)
Comment (ewelinagr) [reply src/app/services/cohort.service.ts]: it is removed from there as well

Comment (gijskant) [reply src/app/services/cohort.service.ts]: ah, sorry, I missed that

**Comment by ewelinagr** [comment]
thanks @gijskant, the problem you mentioned is fixed now.

**Comment by gijskant** [comment]
The discrepancy now seems to be the other way around: when its is not disabled when dragging in (e.g., type from 100 cats), it is disabled after restore.

**Comment by ewelinagr** [comment]
> The discrepancy now seems to be the other way around: when its is not disabled when dragging in (e.g., type from 100 cats), it is disabled after restore.

this is actually done on purpose, but let's discuss the ticket tomorrow then.

**Pull request merged**